iq/iq.go

```
@@ -0,0 +1,221 @@
+		nil,
+	)
+
+	req.SetBasicAuth(localConfig.User, localConfig.Token)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		var response applicationResponse
+		json.Unmarshal(bodyBytes, &response)
+		return response.Applications[0].ID
+	}
+	return ""
+}
+
+func submitToThirdPartyAPI(sbom string, internalID string) string {
+	client := &http.Client{}
+
+	url := fmt.Sprintf("%s%s", localConfig.Server, fmt.Sprintf("%s%s%s%s", thirdPartyAPILeft, internalID, thirdPartyAPIRight, localConfig.Stage))
+
+	req, err := http.NewRequest(
+		"POST",
+		url,
+		bytes.NewBuffer([]byte(sbom)),
+	)
+
+	req.SetBasicAuth(localConfig.User, localConfig.Token)
+
+	req.Header.Set("Content-Type", "application/xml")
+	req.Header.Set("User-Agent", fmt.Sprintf("nancy-client/%s", buildversion.BuildVersion))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusAccepted {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		var response thirdPartyAPIResult
+		json.Unmarshal(bodyBytes, &response)
+		return response.StatusURL
+	}
+
+	return ""
+}
+
+func pollIQServer(statusURL string, finished chan bool) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", statusURL, nil)
+
+	req.SetBasicAuth(localConfig.User, localConfig.Token)
+
+	req.Header.Set("User-Agent", fmt.Sprintf("nancy-client/%s", buildversion.BuildVersion))
+
+	resp, err := client.Do(req)
+
+	if err != nil {
+		finished <- true
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Print(err)
```

If this fails should we exit??

---

I was on the fence about that. Probably. I was just worried that you start polling and maybe there is a quick network blip? I actually wanted to just do a like "120 max tries" approach, or something like that.

---

So then swallow the error?? Or at least collect it and print it only if it truly fails?? Feels like noise if its really just a retry.

---

I went this route: https://github.com/sonatype-nexus-community/nancy/pull/68/commits/93eceafaf5ae40deea8e64c0c58ad36bc9be696e